pkg/resource/plugin/analyzer_plugin.go

```diff
@@ -82,16 +83,19 @@ func NewPolicyAnalyzer(
 	if err != nil {
 		return nil, rpcerror.Convert(err)
 	} else if pluginPath == "" {
-		return nil, workspace.NewMissingError(workspace.PluginInfo{
-			Kind: workspace.AnalyzerPlugin,
-			Name: string(name),
-		})
+		return nil, fmt.Errorf("could not start policy pack %s because the built-in analyzer "+
```

Comment:
The phrase "the built-in analyzer ... is missing" doesn't sound right. If it was built-in, how could it be missing? (i.e. implying it is part of the `pulumi` CLI or installation.)

Is the end user supposed to know what "the plugin directory" is? I would assume not, and so they wouldn't know how to try and act on that suggestion.

Similarly, "the Pulumi SDK does not support resource policies"? I assume that is referring to the Pulumi libraries downloaded as part of the program, e.g. `package.json`? Can we clarify this somehow, since I don't believe "Pulumi SDK" is a well-understood concept. Perhaps, "the version of the Pulumi libraries used with this application are too old"?

Reply:
Hmm, I'm not sure how to turn this feedback into a more useful error message. Do you have an error message in mind?

Overall the philosophy for this error message was this: we don't expect a user to hit this situation at all, and if they do, we can't remediate it. So, we might as well just say what we think went wrong, so they can at least Google it.

If we think we can be more helpful while describing what went wrong in less detail, I'm all for it, but I happen to think that is a tall order.

Reply:
Why do we need to change this from what `NewMissingError` does?  If we do need to change it, any reason not to just create a very similar message?  That previous message seemed to address these questions well and makes more sense to be consistent than not.

Reply:
We changed the error because it's useless for this case.  In that scenario, when you do a `pulumi policy publish hausdorff/foo`, and the `pulumi-analyzer-policy` script (which should be built in) is missing, you get an error message like `could not find analyzer plugin 'foo' on your $PATH`. That's worse than useless—it's just wrong. The actual problem is what we say in this error.

Reply:
From your message, @hausdorff it sounds like this code is trying to handle the case where we can't load the `pulumi-analyzer-policy` is missing.  Could we have just continued to return NewMissingError here and instead of setting Name to `string(name)` just pass in `policyAnalyzerName`?  Wouldn't that lead to an error message saying it couldn't find `pulumi-analyzer-policy` on your `$PATH` (and that's pretty actionable?)

Reply:
Users who are running old versions of the CLI/SDK are going to run into this problem, and that's probably going to be a nontrivial number of them. Is this issue that we don't think that this error message actually tells them useful information, or that we really want it to use `NewMissingError`?